src/app.js

```diff
@@ -141,6 +135,86 @@ go.app = function() {
             });
         });
 
+        self.states.add('states:display-data', function(name, opts) {
+            var section_data = opts.opts_data[opts.section_id]; 
+            var return_text = sub_section(section_data, opts.section_id);
+            return new ChoiceState(name, {
+                question: 'You have chosen to query ' + opts.section_name + ' in ' + capitaliseLocation(opts.location_input),
+
+                choices: [
+                    new Choice('states:sms', 'SMS details to me'),
+                    new Choice('states:retrieve-location', 'Query another section'),
+                    new Choice('states:start', 'Main Menu'),
+                    new Choice('states:end', 'Exit')],
+
+                next: function(choice) {
+                    if (choice.value == 'states:start' || choice.value == 'states:end') {
+                        return choice.value;
+                    } else if (choice.value == 'states:retrieve-location'){
+                        return {
+                            name: choice.value,
+                            creator_opts: {
+                                full_geoid : opts.location_id,
+                                full_name: opts.location_name,
+                                location_input : opts.location_input
+                            }
+                        };
+                    } else if (choice.value == "states:sms") {
+                        return {
+                            name: choice.value,
+                            creator_opts: {
+                                location_input: opts.location_input,
+                                return_text : return_text,
+                                section_name : opts.section_name,
+                                location_id : opts.location_id 
```

Comment:
We don't seem to be using this field for the SMS state? Unless I'm missing something.